### PR TITLE
feat: Added functionality for creating new streams and consumers in nats

### PR DESCRIPTION
**Pull request by akshedu**

### CODE REVIEW

**Comment by uknth** [comment]
Right of the bat, without even reading your code. 
1. `feat: <information>` if you are writing new feature, feat is enough, you don't need to explain "added functionality", you could have written `feat: creates new stream and consumers in nats`
2. Commit messages are imperative, which basically means you are giving an instruction. So, instead of `Added` use verbs like `adds` etc.

File: kit/transport/nats/consumer.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	}
+
+	c.js = js
+	c.conn = cc
+	c.info = info
+
+	return c, nil
+}
+
+func (c *Consumer) GetConsumerInfo() (*natn.ConsumerInfo, error) {
```
Comment (uknth) [inline kit/transport/nats/consumer.go:192]: Can you use an internal object to `go-base` as output of this method? This leaks the internal details which will then require the upstream to import nats dependency.

File: kit/transport/nats/consumer.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	return func(c *Consumer) {
+		c.cfg.InactiveThreshold = inactiveThreshold
+	}
+}
+
+func defaultConsumerErrorHandler(cx context.Context, err error) error {
+	return err
+}
+
+func NewConsumer(connstr string, stream string, options ...ConsumerOption) (*Consumer, error) {
```
Comment (uknth) [inline kit/transport/nats/consumer.go:146]: Can you keep this in line with rest of the NATS implementation?
Either change NATS to call it Consumer everywhere, i.e. existing implementation as well or change this to `NewStreamSubscriber()`

File: kit/transport/nats/consumer.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+	cc, err = c.opts.Connect()
+
+	if err != nil {
+		return nil, errors.Wrap(
+			err, "unable to connect to nats server",
+		)
+	}
+
+	// Create JetStream Context
+	js, _ := cc.JetStream()
```
Comment (uknth) [inline kit/transport/nats/consumer.go:176]: If the second variable returned is an error, handle it.

**Review by uknth — CHANGES_REQUESTED** [review]
The way I understood what you are doing here is, you will do something like this in your application
```
si, err := stream.GetInfo()
// check err

si.SomeMethod() // where some method is exposed by the NATS library
```
If this is the case, your implementation isn't correct. You are leaking information. Understand this, the whole idea of this project is to have `endpoint.Endpoint` be central exposed method. This is to ensure that we can change the implementation from NATS to HTTP or KAFKA very easily. 

This current implementation isn't really a library, it is more like you directly using NATS in your application, just your code is residing here.

**Comment by keshareenv** [comment]
@uknth @akshedu Can we take this forward please, rn we are using the fork for deadpool initial release

**Comment by akshedu** [comment]
@uknth let me know if any changes are required as per the latest implementation. After our discussion I had incorporated everything

**Pull request closed without merging**